general/cuda.hpp

```diff
@@ -77,6 +77,18 @@ void mfem_cuda_error(cudaError_t err, const char *expr, const char *func,
                      const char *file, int line);
 #endif
 
+template <typename T>
+MFEM_HOST_DEVICE T AtomicAdd(T &add, const T val)
+{
+#if defined(MFEM_USE_CUDA) && defined(__CUDA_ARCH__)
```

Comment:
Minor comment, how far are we from supporting AMD GPUs? Might be good add the hip equivalent of this.

Reply:
We already do, there's this branch that will allow us to get right level of performances:
https://github.com/mfem/mfem/blob/hip-defs/general/hip.hpp